### PR TITLE
[RJSF] Add a Custom Object Field to fix validation errors | CU-3u6yyf2

### PR DESCRIPTION
**Pull request by humblenginr**
**Description**

Current Scenario:
Whenever RJSF sees the type `object` in the schema, it will use the `ObjectField` component to render that property. 
The `ObjectField` component uses the `ObjectFieldTemplate` to allow the users to add modifications to the layout.

This PR adds a custom RJSF Object Field so that the _object_ type level errors are propagated to the `ObjectFieldTemplate` component

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
Task linked: [CU-3u6yyf2 [Designer] Overzealous Design Validation](https://app.clickup.com/t/3u6yyf2)

**Comment by sudo-NithishKarthik** [comment]
Editing the PR description...

File: ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/CustomObjectField.js [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+                name,
+                addedByAdditionalProperties
+              )}
+              onBlur={onBlur}
+              onFocus={onFocus}
+              registry={registry}
+              disabled={disabled}
+              readonly={readonly}
+              hideError={hideError}
+              onDropPropertyClick={this.onDropPropertyClick}
```
Comment (leecalcote) [inline ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/CustomObjectField.js:304]: This is a custom field-specific to the Meshery project?

Comment (sudo-NithishKarthik) [reply ui/components/MesheryMeshInterface/PatternService/RJSFCustomComponents/CustomObjectField.js:304]: A bit confusing, I agree.

**Review by leecalcote — APPROVED** [review]

**Pull request merged**